cmd/bosun/sched/check.go

```diff
@@ -185,9 +186,63 @@ func (s *Schedule) RunHistory(r *RunHistory) {
 	if checkNotify && s.nc != nil {
 		s.nc <- true
 	}
+	s.CollectStates()
 	s.Save()
 }
 
+// CollectStates sends various state information to bosun with collect.
+func (s *Schedule) CollectStates() {
+	metadata.AddMetricMeta(
+		"bosun.alerts.current_severity", metadata.Gauge, metadata.Alert,
+		"The number of open alerts by current severity.")
+	metadata.AddMetricMeta(
+		"bosun.alerts.last_abnormal_severity", metadata.Gauge, metadata.Alert,
+		"The number of open alerts by last abnormal severity.")
+	metadata.AddMetricMeta(
+		"bosun.alerts.acknowledgement_status", metadata.Gauge, metadata.Alert,
+		"The number of open alerts by acknowledgement status.")
+	metadata.AddMetricMeta(
+		"bosun.alerts.active_status", metadata.Gauge, metadata.Alert,
+		"The number of open alerts by active status.")
+	for _, state := range s.status {
+		if !state.Open {
+			return
+		}
+		// The tagset of the alert is not included because there is no way to
+		// store the string of a group in OpenTSBD in a parsable way. This is
+		// because any delimiter we chose could also be part of a tag key or tag
+		// value.
+		ts := opentsdb.TagSet{"alert": state.Alert}
```

Comment:
"Won't this cause data with the same tagset but different values?" Yes that was pretty stupid, think I will map[alert.name]int64 for the counts, and init all alertnames to zero.